### PR TITLE
AO3-6566 ARIA controls must contain the exact ID

### PR DESCRIPTION
**Pull request by neuroalien**
# Pull Request Checklist

* [x] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [x] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [ ] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [x] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [x] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6566

## Purpose

What does this PR do?

Correctly declare the ID of the element controlled by the link to modal.

The ID of the modal controlled by the link to modal is `modal`. The hash sign must not be included in the reference to the element.

## Testing Instructions

How can the Archive's QA team verify that this is working as you intended?

If you have a Jira account with access, please update or comment on the issue
with any new or missing testing instructions instead.

## References

Are there other relevant issues/pull requests/mailing list discussions?

## Credit

What name and pronouns should we use to credit you in the [Archive of Our Own's Release Notes](https://archiveofourown.org/admin_posts?tag=1)?

If you have a Jira account, please include the same name in the "Full name"
field on your Jira profile, so we can assign you the issues you're working on.

*Please note that if you do not fill in this section, we will use your GitHub account name and
they/them pronouns.*

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -150,7 +150,7 @@ def link_to_modal(content = "", options = {})
     options[:for] ||= ""
     options[:title] ||= options[:for]
 
-    html_options = { "class" => options[:class] + " modal", "title" => options[:title], "aria-controls" => "#modal" }
+    html_options = { "class" => "#{options[:class]} modal", "title" => options[:title], "aria-controls" => "modal" }
```
Comment (sarken) [inline app/helpers/application_helper.rb:153]: Does this still need the old hash rocket syntax or does `html_options = { class: "#{options[:class]} modal", title: options[:title], aria: { controls: "modal" } }` work?

And it's okay if you don't know the answer to this, but any idea why we set the `aria-controls` attribute here and in the JavaScript? It seems like the JavaScript is overwriting this: if I just change it here and I check the modals in, e.g., the symbols block on blurbs or anywhere on the search pages, the modal links still have whatever `aria-controls` is set in the JavaScript.

**Review by sarken — COMMENTED** [review]
Question/suggestion, but otherwise looks good!

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (neuroalien) [reply app/helpers/application_helper.rb:153]: I'm pretty sure there's no need for the old syntax for the first two keys, but I didn't know you could nest aria keys like that. I also wasn't sure if we wanted to do piecemeal syntax updates, or if that would open a whole can of worms. 😅 

Is it meant to work without JavaScript? No clue. 🤷

Comment (sarken) [reply app/helpers/application_helper.rb:153]: Without JavaScript, the modal definitely won't work -- it should just fall back to being a plain link, but I haven't tested it to see if that's wishful linking -- so including the ARIA attribute is pretty much unnecessary.

Comment (neuroalien) [reply app/helpers/application_helper.rb]: Updated!

**Review by sarken — APPROVED** [review]

**Pull request merged**